components/eamxx/src/dynamics/homme/interface/homme_driver_mod.F90

```diff
@@ -242,11 +242,16 @@ subroutine prim_run_f90 (nsplit_iteration) bind(c)
   end subroutine prim_run_f90
 
   subroutine prim_finalize_f90 () bind(c)
-    use homme_context_mod,    only: is_model_inited, elem, dom_mt, close_homme_log
+    use homme_context_mod,    only: is_model_inited, elem, par, dom_mt, close_homme_log
     use prim_cxx_driver_base, only: prim_finalize
 
     if (.not. is_model_inited) then
-      call abortmp ("Error! prim_init_model_f90 was not called yet (or prim_finalize_f90 was already called).\n")
+      if (par%masterproc) then
```

Comment:
@brhillman @ndkeen this should fix the red herring error message you (and probably others) observed. SInce the error happened during the stack unwinding while handling a thrown exception, we were completely hiding the actual error message. I verified that this fix brings back error messages.